stan/math/prim/prob/gaussian_dlm_obs_lpdf.hpp

```diff
@@ -112,15 +112,8 @@ gaussian_dlm_obs_lpdf(
     Eigen::Matrix<T_lp, Eigen::Dynamic, 1> m(n);
     Eigen::Matrix<T_lp, Eigen::Dynamic, Eigen::Dynamic> C(n, n);
 
-    // TODO(anyone): how to recast matrices
-    for (int i = 0; i < m0.size(); i++) {
-      m(i) = m0(i);
-    }
-    for (int i = 0; i < C0.rows(); i++) {
-      for (int j = 0; j < C0.cols(); j++) {
-        C(i, j) = C0(i, j);
-      }
-    }
+    m = m0;
```

Comment:
Line 112 should have all the logic in it:

```
Eigen::Matrix<T_lp, -1, 1> m{m0};
```

This should work if they obeyed the rule of three.  If not, use a compound declare/assign:

```
Eigen::Matrix<T_lp, -1, 1> m = m0;
```

Also, the same should be done for `C`.

Reply:
Yeah definitely want to combine those lines, thank you. And direct-list-initialization does work here. 

In this case, direct-initialization, m(m0), would have the same effect; since Eigen::Matrix isn't aggregate, and there are no std::initializer_list constructors or narrowing conversions in sight. Do we prefer direct-list-initialization over direct-initialization when both are possible? Because lines 114-121 (among others) could be using direct-list-initialization instead as well. It looks a little odd to use braces just for C and m.

```
    Eigen::Matrix<return_type_t<T_y>, Eigen::Dynamic, 1> yi{r};
    Eigen::Matrix<T_lp, Eigen::Dynamic, 1> a{n};
    Eigen::Matrix<T_lp, Eigen::Dynamic, Eigen::Dynamic> R{n, n};
    ...
```

Reply:
So far we haven't developed a standard.  I thought the standard practice in C++ was going to be curly braces for constructors, but I have no idea why I thought that.

Reply:
O I'm not disagreeing! I don't know what to do with the whole braces thing and I was hoping their was a policy. Constructors in C++ are kind of weird, they pretend to be functions but then do a bunch of special stuff, so it makes a certain amount of sense for them to just get their own syntax.

For what it's worth, the "cpp core guidelines" have this to say:
"Prefer {}. The rules for {} initialization are simpler, more general, less ambiguous, and safer than for other forms of initialization."

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list

So I think you're right; braces for constructors. And just fix up old code whenever we touch it.